pkg/services/settings/settings.go

```diff
@@ -0,0 +1,43 @@
+package settings
+
+import "time"
+
+// Provider is a settings provider abstraction
+// with thread-safety and runtime updates.
+type Provider interface {
+	// Refresh
+	Refresh() error
+	// KeyValue returns a key-value abstraction
+	// for the given pair of section and key.
+	KeyValue(section, key string) KeyValue
+	// Section returns a settings section
+	// abstraction for the given section name.
+	Section(section string) Section
+}
+
+// Section is a settings section copy
+// with all of its pairs of keys-values.
+type Section interface {
+	// KeyValue returns a key-value
+	// abstraction for the given key.
+	KeyValue(key string) KeyValue
+}
```

Comment:
I'm not strongly in favor of this, because it may cause confusion since it would be a "constant" copy of the keys and values of that section, so, someone could store it by mistake and no longer read new updates. However, the code looks uglier if you need to call `KeyValue` by providing the same `section` every time, so I decided to keep it with a comment to make it explicit.

But I'm still doubtful... 🤔

Reply:
users should be able to store a "Section" and repeatably call it to get the current data.  see https://github.com/grafana/grafana/pull/32219/files#r599041781 for details

Reply:
Agreed.